From bbf591c6c2d7d1dc9287db0728e3e821ba9b7666 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Milan=20P=2E=20Stani=C4=87?= <mps@arvanta.net>
Date: Thu, 30 Apr 2020 16:23:30 +0200
Subject: [PATCH] fix missing '#include <sys/stat.h>' in
 'src/include/liburing.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

make failed on musl libc system when entering test subdir with this:
-----
make[1]: Entering directory '/work/devel/liburing/test'
     CC poll
In file included from poll.c:12:
/usr/include/sys/poll.h:1:2: warning: #warning redirecting incorrect #include <sys/poll.h> to <poll.h> [-Wcpp]
    1 | #warning redirecting incorrect #include <sys/poll.h> to <poll.h>
      |  ^~~~~~~
In file included from poll.c:15:
../src/include/liburing.h:339:35: error: unknown type name 'mode_t'
  339 |      const char *path, int flags, mode_t mode)
      |                                   ^~~~~~
make[1]: *** [Makefile:39: poll] Error 1
make[1]: Leaving directory '/work/devel/liburing/test'
make: *** [Makefile:13: all] Error 2
-----

adding #include <sys/stat.h> to src/include/liburing.h fixes build

Signed-off-by: Milan P. StaniÄ‡ <mps@arvanta.net>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 src/include/liburing.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/include/liburing.h b/src/include/liburing.h
index 56e32d7..dd85f7b 100644
--- a/src/include/liburing.h
+++ b/src/include/liburing.h
@@ -8,6 +8,7 @@ extern "C" {
 
 #include <sys/socket.h>
 #include <sys/uio.h>
+#include <sys/stat.h>
 #include <signal.h>
 #include <inttypes.h>
 #include <time.h>
-- 
2.26.2

